library/network/src/modules/NetworkService.rb

```diff
@@ -93,18 +94,15 @@ def main
     # Helper to run systemctl actions
     # @return exit code
     def RunSystemCtl(service, action)
-      cmd = Builtins.sformat("%1 %2 %3.service", SYSTEMCTL, action, service)
-      ret = Convert.convert(
-        SCR.Execute(path(".target.bash_output"), cmd, "TERM" => "raw"),
-        from: "any",
-        to:   "map <string, any>"
-      )
+      cmd = Builtins.sformat("%1 %2 %3.service", SYSTEMCTL.shellescape,
+        action.shellescape, service.shellescape)
+      ret = SCR.Execute(path(".target.bash_output"), cmd, "TERM" => "raw")
       Builtins.y2debug("RunSystemCtl: Command '%1' returned '%2'", cmd, ret)
       Ops.get_integer(ret, "exit", -1)
     end
 
     def run_wicked(*params)
-      cmd = "#{WICKED} #{params.join(" ")}"
+      cmd = "#{WICKED} #{params.map(&:shellescape).join(" ")}"
```

Comment:
Same as above: That const is used only once, and it obscurs what's going on here. I suggest to change this to a plain string as well:

```Ruby
cmd = "/usr/bin/wicked..."
```